### PR TITLE
Update Location class and enums

### PR DESCRIPTION
**Pull request by bderusha**
## Summary
Updates to Location class needed for Story#209 but pulled out separately to ensure alignment with other workstreams

## Changes

- List of comprehensive changes

## Checklist

- [X] Local Tests Passing?
- [ ] CICD and Pipeline Tests Passing?
- [ ] Added any new Tests?
- [ ] Documentation Updates Made?
- [ ] This is not a breaking change. If it is, please describe it below.

## Is this a breaking change?
If yes, what workflow does this break?
Would break existing workflows for CloudProvider locations (of which there a presently none)

### CODE REVIEW

File: src/dotnet/CarbonAware/Model/Location.cs [hunk 1]
```diff
@@ -19,16 +23,18 @@ public class Location
     /// </summary>
     public decimal? Longitude { get; set; }
 
+
     /// <summary>
-    /// Gets or sets the type of location this location object represents.
+    /// Gets or sets the cloud provider name to use.  When loation type is GeoPosition, this value should be null..
```
Comment (juzuluag) [inline src/dotnet/CarbonAware/Model/Location.cs:28]: nit: typo `loation`

File: src/dotnet/CarbonAware/Model/Location.cs [hunk 2]
```diff
@@ -19,16 +23,18 @@ public class Location
...
     /// </summary>
-    public LocationType LocationType { get; set; } = LocationType.NotProvided;
+    public CloudProvider? ProviderName { get; set; }
```
Comment (juzuluag) [inline src/dotnet/CarbonAware/Model/Location.cs:30]: Should be the variable be renamed to something else? rather than `ProviderName`?

**Review by juzuluag — APPROVED** [review]
Two small comments. Other than that lgtm.

File: src/dotnet/CarbonAware/Model/Location.cs (hunk 2 above)
Comment (bderusha) [reply src/dotnet/CarbonAware/Model/Location.cs:30]: Suggestions?  Should it just match the enum and be `CloudProvider` ?

```
Location.LocationType // CloudProvider
Location.ProviderName // Azure
Location.RegionName // eastus
```
VS

```
Location.LocationType // CloudProvider
Location.CloudProvider // Azure
Location.RegionName // eastus
```

**Review by rakker91 — APPROVED** [review]

**Pull request merged**